server/src/main/java/org/elasticsearch/action/admin/indices/shrink/ResizeRequest.java

```diff
@@ -45,17 +46,22 @@
 public class ResizeRequest extends AcknowledgedRequest<ResizeRequest> implements IndicesRequest, ToXContentObject {
 
     public static final ObjectParser<ResizeRequest, Void> PARSER = new ObjectParser<>("resize_request");
+    private static final ParseField MAX_SINGLE_PRIMARY_SHARD_SIZE = new ParseField("max_single_primary_shard_size");
     static {
         PARSER.declareField((parser, request, context) -> request.getTargetIndexRequest().settings(parser.map()),
             new ParseField("settings"), ObjectParser.ValueType.OBJECT);
         PARSER.declareField((parser, request, context) -> request.getTargetIndexRequest().aliases(parser.map()),
             new ParseField("aliases"), ObjectParser.ValueType.OBJECT);
+        PARSER.declareField(ResizeRequest::setMaxSinglePrimaryShardSize,
+            (p, c) -> ByteSizeValue.parseBytesSizeValue(p.text(), MAX_SINGLE_PRIMARY_SHARD_SIZE.getPreferredName()),
+            MAX_SINGLE_PRIMARY_SHARD_SIZE, ObjectParser.ValueType.STRING);
     }
 
     private CreateIndexRequest targetIndexRequest;
     private String sourceIndex;
     private ResizeType type = ResizeType.SHRINK;
     private Boolean copySettings = true;
+    private ByteSizeValue maxSinglePrimaryShardSize;
```

Comment:
This class is missing serialization of this value (with version checks) in the `StreamInput` constructor and `writeTo` method

Reply:
I also notice this class is missing a `hashCode` and `equals` implementation, which means that we can never compare them. We should remedy that, but it can be done in subsequent work (doesn't have to be done in this PR)

(this is not caught because the tests need to be enhanced to check for transport serialization, that can be done as the subsequent work also)

Reply:
I have added some serialization code in the `StreamInput ` constructor and `writeTo` method yet, and  
I'm glad to open another PR to add the missing `hashCode` and `equals` method.